lib/node/runner/block_operations.go

```diff
@@ -26,12 +27,61 @@ func NewSavingBlockOperations(st *storage.LevelDBBackend, logger logging.Logger)
 	}
 
 	logger = logger.New(logging.Ctx{"m": "SavingBlockOperations"})
-	return &SavingBlockOperations{
-		st:           st,
-		log:          logger,
-		saveBlock:    make(chan block.Block, 10),
-		checkedBlock: common.GenesisBlockHeight,
+	sb := &SavingBlockOperations{
+		st:        st,
+		log:       logger,
+		saveBlock: make(chan block.Block, 10),
 	}
+	sb.checkedBlock = sb.getCheckedBlock()
+	sb.log.Debug("last checked block is", "height", sb.checkedBlock)
+
+	return sb
+}
+
+func (sb *SavingBlockOperations) getCheckedBlockKey() string {
+	return fmt.Sprintf("%s-last-checked-block", common.TmpPrefix)
+}
+
+func (sb *SavingBlockOperations) getCheckedBlock() uint64 {
+	if found, err := sb.st.Has(sb.getCheckedBlockKey()); err != nil {
+		sb.log.Error("failed to check CheckedBlock", "error", err)
+		return common.GenesisBlockHeight
+	} else if !found {
+		return common.GenesisBlockHeight
+	}
+
+	var checked uint64
+	if err := sb.st.Get(sb.getCheckedBlockKey(), &checked); err != nil {
+		sb.log.Error("failed to check CheckedBlock", "error", err)
+		return common.GenesisBlockHeight
+	}
+
+	return checked
+}
+
+func (sb *SavingBlockOperations) saveCheckedBlock(height uint64) {
+	sb.log.Debug("save CheckByBlock", "height", height)
+
+	sb.checkedBlock = height
```

Comment:
I think that this statement is better to be placed at the bottom of this function to make it consistent with the database.